Feature/milestonev1

aracelicaes opened the pull request:
TSE, please ensure you have Ruby Version 2.7.1 Installed as well as Rails Version 6.0.0 
Run bundle install since we updated project gem versions.
In this Branch, we added the Friendship Model with the 2 foreign keys user_id and friend_is (that makes reference to the User id from the User Class).
We then added the associations of the Friendship Model inside the User Model, along with the methods that validate friendship request, pending request, confirmed request, and the one that evaluates if that friendship exists (#friends?(user)).
We followed this tutorial:
https://explainextended.com/2009/03/07/selecting-friends/

Please let us know if you have any questions.
Jess and Ara.

jessicafarias committed:
Create friendship Model and db:migrate

aracelicaes committed:
Add User Associations and Methods
Add friendships association in User
Add inverse_friendships association in User
Add friends method in User
Add #pending_friends in User
Add #friend_requests in User
Add #confirm_friend in User
Add #friend?(*) in User

aracelicaes committed:
Updated ReadMe with Ruby and Rails new versions

jessicafarias committed:
Fixing rubocop errors

uche-inyama reviewed (changes requested):
## Changes Required

### Hi team :wave: 

The models were implemented correctly.

However, I was not able to make or accept friend requests; do add the following features to your
application:

* [x]  A user can invite friends ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/send_invitation.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

* [x]  A user can see friendships requests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/pending_request.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

* [x] A user has to accept friendship request to start friendship ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
* [x] A user can reject a friendship request ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/accept_reject_friendship.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))
* [x] There is a button to invite user to friendship on users list page and on single user page ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/milestone5/user_can_see_invite_buttton.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

* [x]  Write unit and integration tests ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/final_project/tests.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/49736080/))

#### Add buttons to accept or decline friend requests and render notifications on the webpage when requests are made.

when you are done, request for another code review.

You can reach out on slack @ Inyama, Uchechukwu Henry, for clarifications.

#### ``` Have a great day ``` :slightly_smiling_face: 

``` Uchechukwu Henry Inyama ```

jessicafarias committed:
Adding friendships controller

Adding create verb for friendships
Adding friendships routes
adding requested_friend def for user models
Adding fields to view friendship status

aracelicaes committed:
Add #invitee to User Ctrlr
Add #invitee?(user) to User Ctlr
Updated Links in User #index views
Updated friendship ctrlr

jessicafarias committed:
Adding accept link

Edit update in friendship controller
Edit user model
Adding Friend confirm label

aracelicaes committed:
Adding #destroy in User ctrl
Adding #destroy method in User Ctrlr
Adding #destroy action in Friendship ctlrlr
Adding Delete button to user views
Adding decline button to user views
Adding conditional to user ctlr to only view other users not the current user.

aracelicaes committed:
Created Helpers
Created #friendship_buttons helper
Created #display_users helper
Added helpers to user/show and user/index
Corrected rubocop
Added bootstrap in gemfile
imported bootstrap in custom.scss

jessicafarias committed:
Adding rspec files

Adding testing for friendship model
Adding test for post model
Adding test for user model
Adding rspec in Gemfile

jessicafarias committed:
Adding rspec and styles

Merge branch 'feature/milestonev1' of github.com:jessicafarias/ror-social-scaffold into feature/milestonev1

aracelicaes committed:
Added a couple more tests to Post
Added Validations tests to post

jessicafarias committed:
Change the colors of the buttons

jessicafarias committed:
Adding styles and rspect test

Merge branch 'feature/milestonev1' of github.com:jessicafarias/ror-social-scaffold into feature/milestonev1

aracelicaes committed:
Delete bckcolor Add mr-2 to Accept button

rOluochKe reviewed (approved):
Hello Team!
### Well done, you got it right :clap: The milestone has been approved. Congratulations :boom: :100:
### Merge and submit the milestone. Good luck with the next milestone. 
- Happy hacking!
